src/App.jsx

```diff
@@ -1,33 +1,40 @@
+import { useState } from 'react';
+
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-export const App = () => (
-  <div className="page">
-    <div className="page-content">
-      <div className="box">
-        <div className="field">
-          {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
-          <label htmlFor="search-query" className="label">
-            Search movie
-          </label>
+export const App = () => {
+  const [query, setQuery] = useState('');
+
+  return (
+    <div className="page">
+      <div className="page-content">
+        <div className="box">
+          <div className="field">
+            {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+            <label htmlFor="search-query" className="label">
+              Search movie
+            </label>
 
-          <div className="control">
-            <input
-              type="text"
-              id="search-query"
-              className="input"
-              placeholder="Type search word"
-            />
+            <div className="control">
+              <input
+                type="text"
+                id="search-query"
+                className="input"
+                placeholder="Type search word"
+                onChange={e => setQuery(e.target.value)}
+              />
+            </div>
           </div>
         </div>
-      </div>
 
-      <MoviesList movies={moviesFromServer} />
-    </div>
+        <MoviesList movies={moviesFromServer} query={query} />
```

Comment:
You shouldn't pass the whole array of movies to the MovieList. It's better to implement filtering logic on the highest level and pass the filtered array instead. .The MovieList component's primary responsibility is to render the list of movies based on the provided data. By keeping the filtering logic separate, you maintain a cleaner and more maintainable code structure.
```suggestion
        <MoviesList movies={filteredMovies} />
```